From: Eike Rathke <erack@redhat.com>
Date: Fri, 22 Aug 2014 12:59:47 +0200
Subject: [PATCH] resolved fdo#82936 do not clear the 3D flag on refupdate

(cherry picked from commit 75aad687cb5e63a41303aea568bf68b2b5f15d31)
Reviewed-on: https://gerrit.libreoffice.org/12304
Reviewed-by: Markus Mohrhard <markus.mohrhard@googlemail.com>
Tested-by: Markus Mohrhard <markus.mohrhard@googlemail.com>

Change-Id: I2ae1368a2ffc20e988ff3ab66590a9f98902b862
---
 sc/source/core/tool/token.cxx | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/sc/source/core/tool/token.cxx b/sc/source/core/tool/token.cxx
--- a/sc/source/core/tool/token.cxx
+++ b/sc/source/core/tool/token.cxx
@@ -2847,7 +2847,8 @@ sc::RefUpdateResult ScTokenArray::AdjustReferenceOnMove(
                 }
 
                 rRef.SetAddress(aAbs, rNewPos);
-                rRef.SetFlag3D(b3DFlag);
+                if (b3DFlag)
+                    rRef.SetFlag3D(b3DFlag);
             }
             break;
             case svDoubleRef:
-- 
1.9.1

